include/npcomp/Dialect/Torch/IR/TorchBase.td

```diff
@@ -0,0 +1,31 @@
+//===-------------------------------------------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TORCH_BASE
+#define TORCH_BASE
+
+include "mlir/IR/OpBase.td"
+
+def Torch_Dialect : Dialect {
+  let name = "torch";
+  let cppNamespace = "::mlir::NPCOMP::Torch";
+  let description = [{
+    Top-level dialect for interfacing PyTorch and MLIR.
+
+    This dialect contains types and structural ops that model enough of
+    PyTorch's behavior to allow for easy import/call-out. While not aiming to
+    be completely isomorphic, it is laid out to make conversion in/out
+    systematic for the supported features.
+
+    As a key point, this dialect does not contain any of the "normal" ops (see
```

Comment:
"normal" is unclear. maybe "see the `ATen` dialect for "matmul" and other mathematical operators"